src/deluge/gui/views/performance_session_view.cpp

```diff
@@ -0,0 +1,1931 @@
+				}
+			}
+		}
+	}
+	return false;
+}
+
+/// called when you press <> + back
+/// in param editor, it will clear existing param mappings
+/// in regular performance view or value editor, it will clear held pads and reset param values to pre-held state
+void PerformanceSessionView::resetPerformanceView(ModelStackWithThreeMainThings* modelStack) {
+	for (int32_t xDisplay = 0; xDisplay < kDisplayWidth; xDisplay++) {
+		if (editingParam) {
+			initLayout(layoutForPerformance[xDisplay]);
+		}
+		else if (fxPress[xDisplay].padPressHeld) {
+			//obtain Param::Kind and ParamID corresponding to the column in focus (xDisplay)
+			Param::Kind lastSelectedParamKind = layoutForPerformance[xDisplay].paramKind; //kind;
+			int32_t lastSelectedParamID = layoutForPerformance[xDisplay].paramID;
+
+			if (lastSelectedParamID != kNoSelection) {
+				padReleaseAction(modelStack, lastSelectedParamKind, lastSelectedParamID, xDisplay, false);
+			}
+		}
+	}
+	updateLayoutChangeStatus();
+	renderViewDisplay();
+	uiNeedsRendering(this);
+}
+
+/// resets a single FX column to remove held status
+/// and reset the param value assigned to that FX column to pre-held state
+void PerformanceSessionView::resetFXColumn(ModelStackWithThreeMainThings* modelStack, int32_t xDisplay) {
+	if (fxPress[xDisplay].padPressHeld) {
+		//obtain Param::Kind and ParamID corresponding to the column in focus (xDisplay)
+		Param::Kind lastSelectedParamKind = layoutForPerformance[xDisplay].paramKind; //kind;
+		int32_t lastSelectedParamID = layoutForPerformance[xDisplay].paramID;
+
+		if (lastSelectedParamID != kNoSelection) {
+			padReleaseAction(modelStack, lastSelectedParamKind, lastSelectedParamID, xDisplay, false);
+		}
+	}
+	updateLayoutChangeStatus();
+}
+
+/// check if parameter is stutter
+bool PerformanceSessionView::isParamStutter(Param::Kind paramKind, int32_t paramID) {
+	if ((paramKind == Param::Kind::UNPATCHED_SOUND) && (paramID == Param::Unpatched::STUTTER_RATE)) {
+		return true;
+	}
+	return false;
+}
+
+/// check if stutter is active and release it if it is
+void PerformanceSessionView::releaseStutter(ModelStackWithThreeMainThings* modelStack) {
+	if (isUIModeActive(UI_MODE_STUTTERING)) {
+		padReleaseAction(modelStack, Param::Kind::UNPATCHED_SOUND, Param::Unpatched::STUTTER_RATE,
+		                 lastPadPress.xDisplay, false);
+	}
+}
+
+/// this will set a new value for a parameter
+/// if we're dealing with stutter, it will check if stutter is active and end the stutter first
+/// if we're dealing with stutter, it will change the stutter rate value and then begin stutter
+/// if you're in the value editor, pressing a column and changing the value will also open the sound editor
+/// menu for the parameter to show you the current value in the menu
+/// in regular performance view, this function will also update the parameter value shown on the display
+bool PerformanceSessionView::setParameterValue(ModelStackWithThreeMainThings* modelStack, Param::Kind paramKind,
```

Comment:
This function should work on param values instead of knob pos - will allow greater accuracy in the future (particularly for things like stutter where the knob pos is very coarse)